gax/src/main/java/com/google/api/gax/retrying/ExponentialRetryAlgorithm.java

```diff
@@ -185,10 +230,28 @@ public boolean shouldRetry(TimedAttemptSettings nextAttemptSettings) {
     return true;
   }
 
+  /**
+   * Returns {@code true} if another attempt should be made, or {@code false} otherwise.
+   *
+   * @param context a {@link RetryingContext} that can contain custom {@link RetrySettings} and
+   *     retryable codes. Ignored by this implementation.
+   * @param nextAttemptSettings attempt settings, which will be used for the next attempt, if
+   *     accepted
+   * @return {@code true} if {@code nextAttemptSettings} does not exceed either maxAttempts limit or
+   *     totalTimeout limit, or {@code false} otherwise
+   */
+  public boolean shouldRetry(RetryingContext context, TimedAttemptSettings nextAttemptSettings) {
+    // The RetrySettings from the context are not used here, as they have already been set as the
+    // global settings during the creation of the initial attempt.
+    return shouldRetry(nextAttemptSettings);
+  }
+
   // Injecting Random is not possible here, as Random does not provide nextLong(long bound) method
   protected long nextRandomLong(long bound) {
-    return bound > 0 && globalSettings.isJittered()
-        ? ThreadLocalRandom.current().nextLong(bound)
-        : bound;
+    return nextRandomLong(bound, globalSettings.isJittered());
+  }
+
+  private long nextRandomLong(long bound, boolean withJitter) {
```

Comment:
Is withJitter necessary? Any problems if this always adds jitter?

Reply:
I think it would be confusing, as:
* It would mean a change from the current behavior. It currently uses the jittered setting in `globalSettings`.
* It would mean that we would not respect one of the properties of `RetrySettings` when a client application sets specific `RetrySettings` for an invocation.

Reply:
Here's the thing: no one should have code that depends on no jitter, period. That's not an option anyone needs or can detect.

Reply:
It might make sense to simply revert this to the original code. It doesn't seem like anything has changed, but we should fix this eventually.

Reply:
I've reverted it to the original code.